backend/core/signals.py

```diff
@@ -0,0 +1,27 @@
+import logging
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
+from .models import User, EmailDomain
+
+logger = logging.getLogger(__name__)
+
+
+@receiver(pre_save, sender=User)
+def ensure_that_email_domain_exists(sender, instance, update_fields, **kwargs):
```

Comment:
Is there any benefit here of using a signal instead of overwriting the `save` method of our `User` model?

Signals are really handy when it comes to make uncoupled Django apps react to each other, but they have the downside of creating scattered and less easy to maintain code.

In our case, both `EmailDomain` and `User` models are part of the same app so, overwriting the `User`'s `save` method won't make uncoupled folders coupled. 

I personally tend to avoid signals as much as I can (i.e. when I don't explicitely need them) so that all behaviours related to a model are visible in the model's file. I was also inspired by the authors of the book [Two Scoops of Django](https://www.feldroy.com/books/two-scoops-of-django-3-x) who ecourage, in the version 2.x of the book, using the model's hooks depending on your use case.

Reply:
I have no strong opinion about this.  

I agree that too many signals may lead to surprising behaviour and hard-to-maintain code.  
In this case, overriding `User.save()` might also be a bit confusing, as the side-effect affects another model `EmailDomain`. But I see that this approach was already used for the (unused) `VerifiableEmail`, so I will make the change and remove the signal.